### ISSUE TITLE
BugFix Ripple | SQLAlchemy==1.3.24 | Multi-column expectations throwing exceptions

### ISSUE THREAD

**Issue by BilalZahid22** [fivetran/great_expectations#3557]
**Describe the bug**
This bug is the successor of the bugfix done in #3468 . Issue doesn't seem to be completely resolved in this issue as multi-column expectations are working fine on sqlalchemy==1.4.25 but still throwing exception(different exception than mentioned in #3468 ) on sqlalchemy==1.3.24. 

A stack trace of this new exception is as follows: 
> expect_column_pair_values_to_be_equal raised an exception:
> columns argument to select() must be a Python list or other iterable
> Show exception traceback...
> 
> Traceback (most recent call last):
> File "D:\Work\Repo\venv\lib\site-packages\sqlalchemy\sql\selectable.py", line 3105, in init
> cols_present = bool(columns)
> File "D:\Work\Repo\venv\lib\site-packages\sqlalchemy\sql\elements.py", line 535, in bool
> raise TypeError("Boolean value of this clause is not defined")
> TypeError: Boolean value of this clause is not defined
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
> File "D:\Work\Repo\venv\lib\site-packages\great_expectations\execution_engine\execution_engine.py", line 294, in resolve_metrics
> resolved_metrics[metric_to_resolve.id] = metric_fn(
> File "D:\Work\Repo\venv\lib\site-packages\great_expectations\expectations\metrics\map_metric_provider.py", line 2084, in sqlalchemy_column_pair_map_condition_values
> query = sa.select(
> File "", line 2, in select
> File "", line 2, in init
> File "D:\Work\Repo\venv\lib\site-packages\sqlalchemy\util\deprecations.py", line 139, in warned
> return fn(*args, **kwargs)
> File "D:\Work\Repo\venv\lib\site-packages\sqlalchemy\sql\selectable.py", line 3107, in init
> util.raise(
> File "D:\Work\Repo\venv\lib\site-packages\sqlalchemy\util\compat.py", line 182, in raise_
> raise exception
> sqlalchemy.exc.ArgumentError: columns argument to select() must be a Python list or other iterable
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
> File "D:\Work\Repo\venv\lib\site-packages\great_expectations\validator\validator.py", line 670, in resolve_validation_graph
> self._resolve_metrics(
> File "D:\Work\Repo\venv\lib\site-packages\great_expectations\validator\validator.py", line 759, in _resolve_metrics
> return execution_engine.resolve_metrics(
> File "D:\Work\Repo\venv\lib\site-packages\great_expectations\execution_engine\execution_engine.py", line 298, in resolve_metrics
> raise ge_exceptions.MetricResolutionError(
> great_expectations.exceptions.exceptions.MetricResolutionError: columns argument to select() must be a Python list or other iterable

**To Reproduce**

1. Create expectation expect_column_pair_values_to_be_equal for **snowflake** using sqlalchemy==1.3.24.
2. Exception will occur on creation time of expectation or if its already created then on time of validation.

**Expected behavior**
There should be support for multi-column expectations on sqlalchemy==1.3.24 (As this is the highest version of sqlalchemy that airflow currently supports)

**Environment (please complete the following information):**
 - Operating System: [Windows]
 - Great Expectations Version: [0.13.38]

**Additional context**

**Comment by talagluck** [comment]
Hi @BilalZahid22 - thanks for following up with this! I believe this should be resolved by #3507 which should be merged into `develop` shortly. Would you be up for testing this out locally once this has merged to confirm that this takes care of the issue?

**Comment by BilalZahid22** [comment]
> Hi @BilalZahid22 - thanks for following up with this! I believe this should be resolved by #3507 which should be merged into `develop` shortly. Would you be up for testing this out locally once this has merged to confirm that this takes care of the issue?

Sure I can test it out once it is merged.

**Comment by BilalZahid22** [comment]
@talagluck I did test it locally by applying changes in #3507 on great-expectations==0.13.38 on my local environment . I didn't tested the whole _develop_ branch on my local as change is not merged yet.
But by applying these changes, the expectations are working fine on sqlalchemy=1.3.24. So it seems #3507  will solve the issue

**Comment by talagluck** [comment]
Great - thanks so much for the update, @BilalZahid22! That was just merged into develop, and will be released later this week. Closing this for now.
